### DISCUSSION TITLE
Optimal Approach for Implementing Delete Confirmation Dialog in Data Table

### DISCUSSION THREAD

**Discussion by JiriKas** [shadcn-ui/ui#1842] [Q&A]
Hello everyone, my question is related to the Data Table component, where I have a list of users, for example, and in the last column, there's a button to delete a user. When I click on the delete button, a confirmation dialog is displayed. Currently, I'm handling this by returning a cell with the Dialog component in the "columns" file. Is this approach correct, or is there a better solution? I have a feeling that I might be unnecessarily rendering the Dialog component for each row.

**Comment by BadisKefi** [comment]
everyone is doing it that way and i believe its correct , i recommend you to think about performance when there is a problem not when there is no problem, cause sometimes the framework itself is smart anought to handle this before you even think about it. sometimes this can cause you doubts and fear for no reason.
hope this helped

**Comment by rahul3103** [comment]
Create a global state, on click of delete button, store user info and open a common dialog with this user info. On click of delete make an api call to delete that user with uid.

**Comment by saifwarraich** [comment]
I had a similar use case. I developed a single `Dialog` component and managed it using state management. When the delete button is clicked, I saved the ID of the specific item in the state and opened the delete dialog. I believe this approach avoids rendering multiple dialogs.
